src/libstd/f32.rs

```diff
@@ -1044,17 +1044,24 @@ impl f32 {
     #[stable(feature = "float_bits_conv", since = "1.20.0")]
     #[inline]
     pub fn from_bits(mut v: u32) -> Self {
-        const EXP_MASK: u32   = 0x7F800000;
-        const FRACT_MASK: u32 = 0x007FFFFF;
-        if v & EXP_MASK == EXP_MASK && v & FRACT_MASK != 0 {
-            // While IEEE 754-2008 specifies encodings for quiet NaNs
-            // and signaling ones, certain MIPS and PA-RISC
-            // CPUs treat signaling NaNs differently.
-            // Therefore to be safe, we pass a known quiet NaN
-            // if v is any kind of NaN.
-            // The check above only assumes IEEE 754-1985 to be
-            // valid.
-            v = unsafe { ::mem::transmute(NAN) };
+        if cfg!(not(any(
```

Comment:
Can you add comments here and below noting why we only do this on some platforms?

Reply:
The existing comment seems to address it, no?

Reply:
The existing comment just says why we reset to a "canonical" NaN instead of just masking off the signal bit.

Reply:
"While IEEE 754-2008 specifies encodings for quiet NaNs and signaling ones, certain MIPS and PA-RISC CPUs treat signaling NaNs differently."

Reply:
Err wait, I think I don't follow what you're asking/saying?

Reply:
The documentation to this function states that "due to the implications of Rust's safety promises being uncertain, if the representation of a signaling NaN "sNaN" float is passed to the function, the implementation is allowed to return a quiet NaN instead".

Someone looking at the implementation may wonder why those questions about safety promises are not a problem on these four architectures, but are on others. There should be a comment saying why this adjustment isn't needed.

Reply:
I would instead suggest changing the documentation to be "due to some platforms not conforming to IEEE 758-2008..."

Reply:
I don't understand what IEEE 758-2008 compliance has to do with LLVM treating sNaN as UB or not.